cdap-client/src/main/java/co/cask/cdap/client/MetricsClient.java

```diff
@@ -59,78 +64,159 @@ public MetricsClient(ClientConfig config) {
   /**
-   * Gets the value of a particular metric.
+   * Searches for metrics tags matching the given tags.
    *
-   * @param metric name of the metric
-   * @param context context of the metric
-   * @param groupBy how to group the metrics
-   * @return value of the metric
+   * @param tags the tags to match
+   * @return the metrics matching the given tags
    * @throws IOException if a network error occurred
    * @throws UnauthorizedException if the request is not authorized successfully in the gateway server
    */
-  public MetricQueryResult query(String metric, @Nullable Map<String, String> context,
-                                 @Nullable String start, @Nullable String end, @Nullable String groupBy)
+  public List<MetricTagValue> searchTags(Map<String, String> tags)
     throws IOException, UnauthorizedException {
 
-    URL url = config.resolveURLV3(String.format("metrics/query?metric=%s%s%s%s%s", metric,
-                                                start == null ? "" : "&start=" + start,
-                                                end == null ? "" : "&end=" + end,
-                                                context == null ? "" : "&context=" + contextToPathParam(context),
-                                                groupBy == null ? "" : "&groupBy=" + groupBy));
+    List<String> queryParts = Lists.newArrayList();
+    queryParts.add("target=tag");
+    addTags(tags, queryParts);
+
+    URL url = config.resolveURLV3(String.format("metrics/search?%s", Joiner.on("&").join(queryParts)));
     HttpResponse response = restClient.execute(HttpMethod.POST, url, config.getAccessToken());
-    return ObjectResponse.fromJsonBody(response, MetricQueryResult.class).getResponseObject();
+    ObjectResponse<List<MetricTagValue>> result = ObjectResponse.fromJsonBody(
+      response, new TypeToken<List<MetricTagValue>>() { }.getType());
+    return result.getResponseObject();
+  }
+
+  /**
+   * Searches for metrics matching the given tags.
+   *
+   * @param tags the tags to match
+   * @return the metrics matching the given tags
+   * @throws IOException if a network error occurred
+   * @throws UnauthorizedException if the request is not authorized successfully in the gateway server
+   */
+  public List<String> searchMetrics(Map<String, String> tags)
+    throws IOException, UnauthorizedException {
+
+    List<String> queryParts = Lists.newArrayList();
+    queryParts.add("target=metric");
+    addTags(tags, queryParts);
+
+    URL url = config.resolveURLV3(String.format("metrics/search?%s", Joiner.on("&").join(queryParts)));
+    HttpResponse response = restClient.execute(HttpMethod.POST, url, config.getAccessToken());
+    ObjectResponse<List<String>> result = ObjectResponse.fromJsonBody(
+      response, new TypeToken<List<String>>() { }.getType());
+    return result.getResponseObject();
   }
 
-  public MetricQueryResult query(String metric, Map<String, String> context)
+  /**
+   * Gets the value of the given metrics.
```

Comment:
I think this is fine since the only reason it's another method is that Java doesn't support optional arguments, but they're basically the same method conceptually. It's true that ideally we'd have all the parameters detailed in the javadoc, but I'll need @gsps1's help for that.. the HTTP handler code isn't very transparent about what values are valid for what params.